sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ColumnDefaultValue.java

```diff
@@ -33,25 +34,20 @@
  * data that do not have these new columns.
  */
 @Evolving
-public class ColumnDefaultValue {
-  private String sql;
-  private Literal<?> value;
+public class ColumnDefaultValue extends DefaultValue {
+  private final Literal<?> value;
 
   public ColumnDefaultValue(String sql, Literal<?> value) {
-    this.sql = sql;
-    this.value = value;
+    this(sql, null /* no expression */, value);
   }
 
-  /**
-   * Returns the SQL string (Spark SQL dialect) of the default value expression. This is the
-   * original string contents of the SQL expression specified at the time the column was created in
-   * a CREATE TABLE, REPLACE TABLE, or ADD COLUMN command. For example, for
-   * "CREATE TABLE t (col INT DEFAULT 40 + 2)", this returns the string literal "40 + 2" (without
-   * quotation marks).
-   */
-  @Nonnull
-  public String getSql() {
-    return sql;
+  public ColumnDefaultValue(Expression expr, Literal<?> value) {
+    this(null /* no sql */, expr, value);
+  }
+
+  public ColumnDefaultValue(String sql, Expression expr, Literal<?> value) {
```

Comment:
Hmm, as now it has `Expression`, why we still need `Literal`? Cannot the expression a literal?

Reply:
Here `Literal` means the exists or initial literal. It contains the value of the expression evaluated at the time of adding a column to the table.

Reply:
But basically in `toV2` method, `exprV2` and `literalV2` are same one, no?

Reply:
it is quite confusing.  Can we call value =>  'existsDefault' to align with the ResolveDefaultColumnsUtils constants?

Reply:
I agree the name is a bit confusing, but I'd prefer to do the rename separately. It will also be limited to internal variables as we can't change public methods. I'd also argue the naming of "initial" and "write" defaults in Iceberg makes a bit more sense than "current" and "exists".

Reply:
Okay. Btw, it might be also good to update the document for this class.